protocol/contracts/templegold/TempleGold.sol

```diff
@@ -0,0 +1,331 @@
+        if (stakingAmount > 0) {
+            _mint(address(staking), stakingAmount);
+            staking.notifyDistribution(stakingAmount);
+        }
+
+        uint256 escrowAmount = TempleMath.mulDivRound(params.escrow, mintAmount, DISTRIBUTION_MULTIPLIER, false);
+        if (escrowAmount > 0) {
+            _mint(address(escrow), escrowAmount);
+            escrow.notifyDistribution(escrowAmount);
+        }
+
+        uint256 gnosisAmount = TempleMath.mulDivRound(params.gnosis, mintAmount, DISTRIBUTION_MULTIPLIER, false);
+        if (gnosisAmount > 0) {
+            _mint(teamGnosis, gnosisAmount);
+            /// @notice no requirement to notify gnosis because no action has to be taken
+        }
+        
+        emit Distributed(stakingAmount, escrowAmount, gnosisAmount, block.timestamp);
+    }
+
+    function _getMintAmount(VestingFactor memory vestingFactorCache) private view returns (uint256 mintAmount) {
+        /// @notice first time mint
+        if (lastMintTimestamp == 0) {
+            mintAmount = TempleMath.mulDivRound(MAX_SUPPLY, vestingFactorCache.numerator, vestingFactorCache.denominator, false);
+        } else {
+            mintAmount = TempleMath.mulDivRound((block.timestamp - lastMintTimestamp) * (MAX_SUPPLY - totalSupply()), vestingFactorCache.numerator, vestingFactorCache.denominator, false);
+        }
+        uint256 totalSupplyCache = totalSupply();
+        if (totalSupplyCache + mintAmount > MAX_SUPPLY) {
+            mintAmount = MAX_SUPPLY - totalSupplyCache;
+        }
+    }
+
+    /// @notice Overriden OFT functions
+
+    /**
+     * @dev Executes the send operation.
+     * @param _sendParam The parameters for the send operation.
+     * @param _fee The calculated fee for the send() operation.
+     *      - nativeFee: The native fee.
+     *      - lzTokenFee: The lzToken fee.
+     * @param _refundAddress The address to receive any excess funds.
+     * @return msgReceipt The receipt for the send operation.
+     * @return oftReceipt The OFT receipt information.
+     *
+     * @dev MessagingReceipt: LayerZero msg receipt
+     *  - guid: The unique identifier for the sent message.
+     *  - nonce: The nonce of the sent message.
+     *  - fee: The LayerZero fee incurred for the message.
+     */
+    function send(
+        SendParam calldata _sendParam,
+        MessagingFee calldata _fee,
+        address _refundAddress
+    ) external payable virtual override(IOFT, OFTCore) returns (MessagingReceipt memory msgReceipt, OFTReceipt memory oftReceipt) {
+        /// cast bytes32 to address
+        address _to = _sendParam.to.bytes32ToAddress();
+        /// @dev user can cross-chain transfer to either whitelisted or self
+        if (msg.sender != _to) { revert ITempleGold.NonTransferrable(msg.sender, _to); }
+
+        // @dev Applies the token transfers regarding this send() operation.
```

Comment:
`super.send()` doesn't work because the virtual function is `external` and not public or internal. that's why i copy pasta. I will add dev info on the diff